### PR TITLE
dont waste work when calling ComputeTipSetState in parallel

### PR DESCRIPTION
**Pull request by whyrusleeping**

### CODE REVIEW

File: chain/stmgr/stmgr.go [hunk 1]
```diff
@@ -55,12 +57,37 @@ func (sm *StateManager) TipSetState(ctx context.Context, ts *types.TipSet) (cid.
...
+		}
+	}
 	cached, ok := sm.stCache[ck]
-	sm.stlk.Unlock()
 	if ok {
+		sm.stlk.Unlock()
 		span.AddAttributes(trace.BoolAttribute("cache", true))
 		return cached[0], cached[1], nil
 	}
+	ch := make(chan struct{})
```
Comment (Kubuxu) [inline chain/stmgr/stmgr.go]: Only small nit. would be making this name longer `chWait`, so it is easier to spot in defer lambda bellow.

Comment (Kubuxu) [reply chain/stmgr/stmgr.go]: This is from a golang guidelines I've read, the longer some value is going to be referenced by something, the longer it name should be (so it is easier to spot and memorize).

Comment (magik6k) [reply chain/stmgr/stmgr.go]: I wouldn't block on that

Comment (Kubuxu) [reply chain/stmgr/stmgr.go]: That is why I said it was only a nit.

**Pull request merged**